addons/web_editor/static/tests/link_test.js

```diff
@@ -0,0 +1,458 @@
+            await nextTick();
+            assert.strictEqual(
+                editable.innerHTML,
+                `<p><a href="#" target="_blank">#</a><br></p>`
+            );
+        });
+
+        QUnit.test(
+            "should be able to create link with ctrl+k , and should make link on two existing characters",
+            async function (assert) {
+                const { editor, editable } = onMount();
+                const node = editable.querySelector("p");
+                setSelection(node, 0);
+                insertText(editor, "Hello");
+                setSelection(node, 1, node, 3);
+                await triggerHotkey("control+k");
+                await triggerHotkey("control+k")
+                inputText('input[id="o_link_dialog_url_input"]', "#");
+                editor.document.querySelector(".o_dialog footer button.btn-primary").click();
+                await nextTick();
+                assert.strictEqual(
+                    editable.innerHTML,
+                    `<p>H<a href="#" target="_blank">el</a>lo</p>`
+                );
+            }
+        );
+
+
+        QUnit.module("Typing based");
+
+        QUnit.test("typing www.odoo.com + space should convert to link", async function (assert) {
+            const { editor, editable } = onMount();
+            const node = editable.querySelector("p");
+            setSelection(node, 0);
+            inputText(".odoo-editor-editable p", "www.odoo.com");
+            insertText(editor, " ");
+            editor.clean();
+            await assert.strictEqual(
+                editable.innerHTML,
+                `<p><a href="http://www.odoo.com">www.odoo.com</a> </p>`
+            );
+        });
+
+        QUnit.test("typing odoo.com + space should convert to link", async function (assert) {
+            const { editor, editable } = onMount();
+            const node = editable.querySelector("p");
+            setSelection(node, 0);
+            inputText(".odoo-editor-editable p", "odoo.com");
+            insertText(editor, " ");
+            editor.clean();
+            assert.strictEqual(
+                editable.innerHTML,
+                `<p><a href="http://odoo.com">odoo.com</a> </p>`
+            );
+        });
+
+        QUnit.test(
+            "typing http://odoo.com + space should convert to link",
+            async function (assert) {
+                const { editor, editable } = onMount();
+                const node = editable.querySelector("p");
+                setSelection(node, 0);
+                inputText(".odoo-editor-editable p", "http://odoo.com");
+                insertText(editor, " ");
+                editor.clean();
+                assert.strictEqual(
+                    editable.innerHTML,
+                    `<p><a href="http://odoo.com">http://odoo.com</a> </p>`
+                );
+            }
+        );
+
+        QUnit.test(
```

Comment:
This is pretty much a duplicate of the previous one.
Remove the previous one (odoo.com) and let's keep this one! (google.co.in) :smile: